src/components/StripeCardInput/index.js

```diff
@@ -0,0 +1,197 @@
+import scriptLoader from 'react-async-script-loader';
+import withReduxForm from '../../libs/hocs/withReduxForm';
+import { useTheme, useUniqueIdentifier, useLocale } from '../../libs/hooks';
+import RenderIf from '../RenderIf';
+import Label from '../Input/label';
+import HelpText from '../Input/styled/helpText';
+import ErrorText from '../Input/styled/errorText';
+import StyledContainer from './styled/container';
+import getError from './helpers/getErrror';
+import getCardElementOptions from './helpers/getCardElementOptions';
+import StyledCardInput from './styled/cardInput';
+
+/**
+ * Stripe Card Input component are used for freeform data entry.
+ * @category Form
+ */
+const StripeCardInput = React.forwardRef((props, ref) => {
+    const {
+        apiKey,
+        label,
+        hideLabel,
+        bottomHelpText,
+        error,
+        className,
+        style,
+        disabled,
+        required,
+        locale,
+        onChange,
+        onFocus,
+        onBlur,
+        tabIndex,
+        isScriptLoaded,
+        isScriptLoadSucceed,
+    } = props;
+    const [stripe, setStripe] = useState(null);
+    const cardRef = useRef();
+    const cardElementId = useUniqueIdentifier('card-element');
+    const theme = useTheme().rainbow;
+    const cardElementOptions = useMemo(() => getCardElementOptions(theme, disabled), [
+        disabled,
+        theme,
+    ]);
+    const localeStripe = useLocale(locale);
+    const elementOptions = useMemo(
+        () => ({
+            locale: localeStripe,
+            fonts: [
+                {
+                    cssSrc: 'https://fonts.googleapis.com/css?family=Lato&display=swap',
+                },
+            ],
+        }),
+        [localeStripe],
+    );
+    useEffect(() => {
+        if (isScriptLoaded && isScriptLoadSucceed && window.Stripe) {
+            setStripe(window.Stripe(apiKey));
+        }
+    }, [apiKey, isScriptLoadSucceed, isScriptLoaded]);
+
+    // eslint-disable-next-line consistent-return
+    useEffect(() => {
+        if (stripe) {
+            const cardNode = cardRef.current;
+            const elements = stripe.elements(elementOptions);
+            const card = elements.create('card', cardElementOptions);
+
+            card.mount(cardNode);
+            card.on('change', event => {
+                const stripeCardEvent = {
+                    stripe,
+                    card,
+                    isEmpty: event.empty,
+                    isComplete: event.complete,
+                    cardBrand: event.brand,
+                    error: getError(event.error),
+                };
+                onChange(stripeCardEvent);
+            });
+
+            card.on('focus', onFocus);
+            card.on('blur', onBlur);
+            const handlerBlur = () => card.blur();
+            const handlerFocus = () => card.focus();
+            cardNode.addEventListener('blur', handlerBlur);
+            cardNode.addEventListener('focus', handlerFocus);
```

Comment:
why do you need a `ref` to listen for focus and `blur` ?? it should work on the JSX like onFocus onBlur ... although is this really necessary?? you are can listen for those events in the card element as well, right?

Reply:
I see what you are doing passing through the focus the stripe input inside ... first, the blur logic is wrong .. think about it ... secund that focus dance is not that simple, what happens with (shift + tab) ??